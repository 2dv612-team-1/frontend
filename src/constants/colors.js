export default {
  primary: '#3f51b5',
  secondary: '#e8eaf6',
};
