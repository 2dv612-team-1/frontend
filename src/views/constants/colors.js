export default {
  primary: "#3f51b5",
  secondary: "#e8eaf6",
  gray: "#566172"
};
