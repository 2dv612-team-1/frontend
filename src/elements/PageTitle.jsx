import styled from 'styled-components';
import colors from '../constants/colors';


const PageTitle = styled.h1`
  font-size: 24px;
  font-weight: bold;
  color: ${colors.primary};
  text-align: ${props => (props.center ? 'center' : 'left')}
`;

export default PageTitle;
