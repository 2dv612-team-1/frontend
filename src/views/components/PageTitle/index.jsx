import styled from "styled-components";
import colors from "../../constants/colors";
import fonts from "../../constants/fonts";

const PageTitle = styled.h1`
  font-size: 24px;
  font-weight: bold;
  color: ${colors.primary};
  font-family: ${fonts.secondary};
  text-align: ${props => (props.center ? "center" : "left")};
  text-transform: capitalize;
`;

export default PageTitle;
