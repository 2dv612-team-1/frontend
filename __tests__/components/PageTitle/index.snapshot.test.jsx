import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import "jest-styled-components";
import PageTitle from "../../../src/components/PageTitle";
import colors from "../../../src/constants/colors";

describe("PageTitle", () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<PageTitle {...props}>{props.children}</PageTitle>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      children: undefined,
      center: undefined
    };
    wrapper = undefined;
  });

  describe("when props are undefined", () => {
    it("Should match snapshot", () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });

    it("should not be center", () => {
      const tree = Wrapper();
      expect(tree).toHaveStyleRule("text-align", "left");
    });

    it("should use primary color", () => {
      const tree = Wrapper();
      expect(tree).toHaveStyleRule("color", colors.primary);
    });
  });

  describe("when props are defined", () => {
    it("Should match snapshot", () => {
      props.children = "Text";
      props.center = true;
      expect(toJson(Wrapper())).toMatchSnapshot();
    });

    it("should be center", () => {
      props.children = "Text";
      props.center = true;
      expect(Wrapper()).toHaveStyleRule("text-align", "center");
    });
  });
});
