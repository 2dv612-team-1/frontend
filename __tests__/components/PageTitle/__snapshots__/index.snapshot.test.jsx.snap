// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle when props are defined Should match snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: bold;
  color: #3f51b5;
  text-align: center;
}

<h1
  className="c0"
>
  Text
</h1>
`;

exports[`PageTitle when props are undefined Should match snapshot 1`] = `
.c0 {
  font-size: 24px;
  font-weight: bold;
  color: #3f51b5;
  text-align: left;
}

<h1
  className="c0"
/>
`;
